Mage/src/main/java/mage/abilities/condition/AndCondition.java

```diff
@@ -0,0 +1,38 @@
+package mage.abilities.condition;
+
+import mage.abilities.Ability;
+import mage.game.Game;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Combines conditions to one compound conditon, all conditions must be
+ * true to return true for the compound condtion.
+ *
+ * @author 18ths
+ */
+public class AndCondition implements Condition {
```

Comment:
There are already implemented `CompoundCondition` with same functionality.

Reply:
True! I knew I must have missed something, this condition felt too basic to not be used anywhere. :)